docs/pages/9 - Contrib Modules.md

```diff
@@ -272,36 +272,240 @@ There's an [example project](https://github.com/lihaoyi/cask/tree/master/example
+    ivy"com.typesafe.play::play-logback:${playVersion()}"
+```
+
+Scala test will be setup as the default test framework and the following test dependencies will be 
+added (the actual version depends on the version of Play you are pulling `2.6.x` or `2.7.x`):
+```
+    ivy"org.scalatestplus.play::scalatestplus-play::4.0.1"
+```
+
+In order to have a working `start` command the following runtime dependency is also added: 
+```
+    ivy"com.typesafe.play::play-akka-http-server:${playVersion()}"
+```
+#### Using `PlayApiModule`
+
+The `PlayApiModule` trait behaves the same as the `PlayModule` trait but it won't process .scala
+.html files and you don't need to define the twirlVersion:
 
+```scala
+// build.sc
+import mill._
 // You have to replace VERSION
 import $ivy.`com.lihaoyi::mill-contrib-playlib:VERSION`,  mill.playlib._
 
-object app extends RouterModule {
-// ...
-} 
+
+object core extends PlayApiModule {
+    //config
+    override def scalaVersion= T{"2.12.8"}
+    override def playVersion= T{"2.7.0"}   
+    
+    object test extends PlayTests
+}  
 ``` 
 
-#### Configuration options
+#### Play configuration options
+
+The play modules themselves don't have specific configuration options at this point but the [router 
+module configuration options](#router-configuration-options) and the [Twirl module configuration 
+options](#twirl-configuration-options) are applicable. 
+
+#### Commands equivalence
+
+Mill commands are targets on a named build. For example if your build is called `core`:
+- compile: `core.compile`
+- run: not implemented. It can be emulated with `mill -w runBackground` but this 
+doesn't do any kind of classloading optimization and may be slower than Play's usual `run`. 
+- start: `core.start` or `core.run`
+- test: `core.test`
+- dist: not implemented, however you can use the equivalent `core.assembly` 
+command to get a runnable fat jar of the project. The packaging is slightly different though.
+
+#### Using `SingleModule`
+
+The `SingleModule` trait allows you to have the build descriptor at the same level as the source
+ code on the filesystem. You can move from there to a multi-module build either by refactoring 
+ your directory layout into multiple subdirectories or by using mill's nested modules feature.
+
+Looking back at the sample build definition in [Using PlayModule](#using-playmodule):
+```scala
+// build.sc
+import mill._
+// You have to replace VERSION
+import $ivy.`com.lihaoyi::mill-contrib-playlib:VERSION`,  mill.playlib._
+
 
-  * `def playVersion: T[String]` (mandatory) - The version of play to use to compile the routes file.
+object core extends PlayModule {
+    //config
+    override def scalaVersion= T{"2.12.8"}
+    override def playVersion= T{"2.7.0"}
+    override def twirlVersion= T{"1.4.0"}
+    
+    object test extends PlayTests
+}  
+``` 
+The directory layout was:
+```text
+.
+├── build.sc
+└── core
+    ├── app
+    │   ├── controllers
+    │   └── views
+    ├── conf
+    ├── logs
```

Comment:
Maybe put the `routes` file in here so that it is obvious that it still belongs here and is picked up and handled automatically

Reply:
I filtered the tree to only keep the directories but I guess I can be a bit more detailed for the conf part.